mm2src/coins_activation/src/platform_coin_with_tokens.rs

```diff
@@ -311,15 +317,22 @@ where
         mm_tokens.extend(tokens);
     }
 
-    let activation_result = platform_coin.get_activation_result().await?;
+    let activation_result = platform_coin.get_activation_result(&req.request).await?;
     log::info!("{} current block {}", req.ticker, activation_result.current_block());
 
     if req.request.tx_history() {
-        platform_coin.start_history_background_fetching(
-            ctx.clone(),
-            TxHistoryStorageBuilder::new(&ctx).build()?,
-            activation_result.get_platform_balance(),
-        );
+        match activation_result.get_platform_balance() {
+            Some(initial_balance) =>
+                platform_coin.start_history_background_fetching(
+                    ctx.clone(),
+                    TxHistoryStorageBuilder::new(&ctx).build()?,
+                    initial_balance,
+                ),
+            None => return MmError::err(EnablePlatformCoinWithTokensError::ActivationRequestError(
+                "Tx history fetching requires getting balances on activation, current request parameters: `tx_history`=true, `get_balances`=false"
+                    .to_string(),
+            )),
+        }
```

Comment:
We are giving specific error if `activation_result.get_platform_balance()` is `None` without knowing it's internal no matter what. Is it safe to do this?

Reply:
> Alternatively, I can fetch balances as a first step of getting tx history if they were not fetched on activation, but I thought since it will be fetched anyways including it in the response would be the right approach. What do you think about this?

I proposed this as a solution here https://github.com/KomodoPlatform/atomicDEX-API/pull/1762#issuecomment-1522171608. There will be some duplicated code to achieve this, we should pass the option to `start_history_background_fetching` in this case and request the balance if it was `None`.

Reply:
mmm, I understand. It's good to separate `get_platform_balance` internal logic from where it is used in case we use it in other places, but in this case `get_platform_balance` depends on the activation response/result which in turn depends on the request, it's really hard to seperate those 3.
To solve this, I can early return this error `Tx history fetching requires getting balances on activation, current request parameters: tx_history=true, get_balances=false` when `tx_history=true, get_balances=false`. It will be better but not ideal, since transaction history fetching will still depend on if `get_platform_balance` is `Some`.

Reply:
> Alternatively, I can fetch balances as a first step of getting tx history if they were not fetched on activation.

I decided to go with this solution, but I will make the balance fetching a part of the spawned history loop, this will allow enable with tokens requests to not wait for the balances response in the get history loop and will separate `tx_history` from `get_balances`, which is the right approach.

Reply:
Done